model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/client/JpaClientMapKeycloakTransaction.java

```diff
@@ -16,147 +16,51 @@
-        super(em);
-    }
-
-    @Override
-    public MapClientEntity create(MapClientEntity mapEntity) {
-        JpaClientEntity jpaEntity = (JpaClientEntity) CLONER.from(mapEntity);
-        if (mapEntity.getId() == null) {
-            jpaEntity.setId(UUIDKey.INSTANCE.yieldNewUniqueKey().toString());
-        }
-        jpaEntity.setEntityVersion(SUPPORTED_VERSION_CLIENT);
-        em.persist(jpaEntity);
-        return jpaEntity;
-    }
-
-    @Override
-    public MapClientEntity read(String key) {
-        if (key == null) return null;
-        UUID uuid = UUIDKey.INSTANCE.fromStringSafe(key);
-        if (uuid == null) return null;
-
-        return em.find(JpaClientEntity.class, uuid);
+        super(JpaClientEntity.class, em);
     }
 
     @Override
-    public Stream<MapClientEntity> read(QueryParameters<ClientModel> queryParameters) {
-        JpaClientModelCriteriaBuilder mcb = queryParameters.getModelCriteriaBuilder()
-                .flashToModelCriteriaBuilder(new JpaClientModelCriteriaBuilder());
-
-        CriteriaBuilder cb = em.getCriteriaBuilder();
-        CriteriaQuery<JpaClientEntity> query = cb.createQuery(JpaClientEntity.class);
-        Root<JpaClientEntity> root = query.from(JpaClientEntity.class);
-        query.select(cb.construct(JpaClientEntity.class,
-                root.get("id"),
-                root.get("version"),
-                root.get("entityVersion"), 
-                root.get("realmId"), 
-                root.get("clientId"), 
-                root.get("protocol"), 
-                root.get("enabled")
-        ));
-
-        //ordering
-        if (!queryParameters.getOrderBy().isEmpty()) {
-            List<Order> orderByList = new LinkedList<>();
-            for (QueryParameters.OrderBy<ClientModel> order : queryParameters.getOrderBy()) {
-                switch (order.getOrder()) {
-                    case ASCENDING:
-                        orderByList.add(cb.asc(root.get(order.getModelField().getName())));
-                        break;
-                    case DESCENDING:
-                        orderByList.add(cb.desc(root.get(order.getModelField().getName())));
-                        break;
-                    default:
-                        throw new UnsupportedOperationException("Unknown ordering.");
-                }
-            }
-            query.orderBy(orderByList);
-        }
-
-        if (mcb.getPredicateFunc() != null) query.where(mcb.getPredicateFunc().apply(cb, root));
-
-        return closing(
-                paginateQuery(em.createQuery(query), queryParameters.getOffset(), queryParameters.getLimit())
-                        .getResultStream())
-                .map(c -> new MapClientEntityDelegate(new JpaClientDelegateProvider(c, em)));
+    public Selection<JpaClientEntity> selectCbConstruct(CriteriaBuilder cb, Root<JpaClientEntity> root) {
+        return cb.construct(JpaClientEntity.class, 
+            root.get("id"), 
```

Comment:
missing column "version"

```suggestion
            root.get("id"),
            root.get("version"),
```

Reply:
thanks, resolved.